net-nntp/sabnzbd/files/sabnzbd.initd

```diff
@@ -1,99 +1,71 @@
 #!/sbin/openrc-run
-# Copyright 1999-2014 Gentoo Foundation
+# Copyright 1999-2016 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-RUNDIR=/var/run/sabnzbd
+PIDFILE="/run/sabnzbd/sabnzbd.pid"
 
 depend() {
-    need net
+	need net
 }
 
 get_var() {
-    echo $(sed -n \
-        '/^\[misc]/,/^'$1'/ s/^'$1' = \([[:alnum:].]\+\)[\r|\n|\r\n]*$/\1/p' \
-        "${SABNZBD_CONFIGFILE}")
+	grep -P -o -m 1 "(?<=^${1} = ).*" "${SABNZBD_CONFIGFILE}" || echo 0
 }
 
-get_port() {
-    if [ "$(get_var 'enable_https')" -eq 1 ]; then
-        echo $(get_var 'https_port')
-    else
-        echo $(get_var 'port')
-    fi
+start() {
+	ebegin "Starting SABnzbd"
+
+	checkpath -q -d -o ${SABNZBD_USER}:${SABNZBD_GROUP} -m 0770 "$(dirname "${PIDFILE}")"
+
+	start-stop-daemon \
+		--quiet \
+		--start \
+		--user ${SABNZBD_USER} \
+		--group ${SABNZBD_GROUP} \
+		--pidfile "${PIDFILE}" \
+		--wait 1000 \
+		--exec /usr/bin/sabnzbd \
+		-- \
+		--config-file "${SABNZBD_CONFIGFILE}" \
+		--logging "${SABNZBD_LOGGING}" \
+		--daemon \
+		--pidfile "${PIDFILE}"
+
+	eend $?
 }
 
-get_addr() {
-    local host=$(get_var 'host')
-    local protocol='http'
+stop() {
+	local protocol="http"
+	local host="$(get_var "host")"
+	local port="$(get_var "port")"
 
-    [ "${host}" == "0.0.0.0" ] && host=localhost
-    [ "$(get_var 'enable_https')" -eq 1 ] && protocol='https'
+	if [ $(get_var "enable_https") -eq 1 ]; then
+		protocol="https"
+		port="$(get_var "https_port")"
+	fi
 
-    echo "${protocol}://${host}:$(get_port)"
-}
+	case "${host}" in
+		*:*) host="[${host}]" ;;
+	esac
```

Comment:
SABnzbd+ uses plain IPv6 addresses in the `host` variable, while Wget needs them to be enclosed in square brackets, so this is needed for proper API-based shutdown to happen over IPv6.